library/src/main/java/com/novoda/downloadmanager/lib/Criteria.java

```diff
@@ -0,0 +1,177 @@
+package com.novoda.downloadmanager.lib;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Criteria {
+
+    public static final Criteria ALL = new Criteria(null, null);
+
+    private final String selection;
+    private final String sort;
+    private final String[] selectionArguments;
+
+    Criteria(String selection, String sort, String... selectionArguments) {
+        this.selection = selection;
+        this.sort = sort;
+        this.selectionArguments = selectionArguments;
+    }
+
+    public String getSelection() {
+        return selection;
+    }
+
+    public String[] getSelectionArguments() {
+        return Arrays.copyOf(selectionArguments, selectionArguments.length);
+    }
+
+    public String getSort() {
+        return sort;
+    }
+
+    public static class Builder {
```

Comment:
I didn't know about `Query`.

>The BatchQuery is our public API where we define different filtering options. I don't know if it should be in sync with Criteria.

The easier to learn our APIs the better to maintain for us. This is already a lot to learn for a few sql queries. I prefer the approach without `Criteria` to build queries because it is less API.